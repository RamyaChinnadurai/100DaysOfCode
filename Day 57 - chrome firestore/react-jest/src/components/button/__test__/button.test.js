import React from 'react';
import ReactDOM from 'react-dom';

import Button from '../button';

import { render, cleanup } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import renderer from 'react-test-renderer';

afterEach(cleanup);

it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<Button></Button>, div);
})

it("renders button correctly", () => {
    const { getByTestId } = render(<Button label="Click me please"></Button>)
    expect(getByTestId('button')).toHaveTextContent('Click me please')
})

it("renders button correctly", () => {
    const { getByTestId } = render(<Button label="save"></Button>)
    expect(getByTestId('button')).toHaveTextContent('save')
})

it("matches snapshot", ()=>{
    const tree = renderer.create(<Button label="save"></Button>).toJSON();
    expect(tree).toMatchSnapshot();
})