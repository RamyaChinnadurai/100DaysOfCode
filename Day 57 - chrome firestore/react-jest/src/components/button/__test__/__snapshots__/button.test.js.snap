// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div
  className="button-style"
  data-testid="button"
>
  save
</div>
`;
